tensor2tensor/rl/rl_trainer_lib.py

```diff
@@ -71,22 +73,33 @@ def define_train(hparams, environment_spec, event_dir):
       # Some environments reset environments automatically, when reached done
       # state. For them we shall record only every second episode.
       d = 2 if env_lambda().metadata.get("semantics.autoreset") else 1
-      eval_env_lambda = lambda: gym.wrappers.Monitor(  # pylint: disable=g-long-lambda
+      monitor_env_lambda = lambda: gym.wrappers.Monitor(  # pylint: disable=g-long-lambda
           env_lambda(), event_dir, video_callable=lambda i: i % d == 0)
       eval_env_lambda = (
-          lambda: utils.EvalVideoWrapper(eval_env_lambda()))
+          lambda: utils.EvalVideoWrapper(monitor_env_lambda()))
+    eval_hparams = copy.deepcopy(hparams)
+    eval_hparams.simulated_environment = eval_hparams.simulated_eval_environment
     eval_batch_env = utils.batch_env_factory(
-        eval_env_lambda, hparams,
+        eval_env_lambda, eval_hparams,
         num_agents=hparams.num_eval_agents, xvfb=hparams.video_during_eval)
 
     # TODO(blazej0): correct to the version below.
-    corrected = False
+    corrected = True
     eval_summary = tf.no_op()
     if corrected:
       _, eval_summary = collect.define_collect(
           policy_factory, eval_batch_env, hparams, eval_phase=True)
   return summary, eval_summary
 
+def parse_from_summary(summary, name):
```

Comment:
I'm not very proud about this function, are there more canonical methods for extracting some values from summaries?